### ISSUE TITLE
Should be able to specify default collation for a new database

### ISSUE THREAD

**Issue by multani** [saltstack-formulas/mysql-formula#78]
`salt.modules.mysql.db_create` has a `collate` attribute which allows to specify a default collation to be used when creating a new database.
It would be nice if this formula allows to specify it.

Problem: the current Pillar configuration doesn't allow to specify such kind of settings:

```
mysql:
  database:
    - foo
    - bar
```

as this is just a list of database names.

If you have a suggestion on how to configure this, I'm willing to submit a pull request for this feature.

**Comment by puneetk** [comment]
Change the pillar from 
```
mysql:
  database:
    - foo
    - bar
```
to support 
```
mysql:
  database:
    - { name: foo, collation: utf8 }
    - bar
```
And then where the pillar is consumed use jinja to check if teh value is a mapping and setup the state accordingly.

**Comment by 0xf10e** [comment]
Lists of dictionaries and lists containing dictionaries can get confusing easily.
Doesn't a database always has some attributes attached?
I would group the stuff specific to the DB under `pillar[mysql:database:$DB_NAME]` and add support for pillar[mysql:database:$DB_NAME:collation]`.

**Comment by iggy** [comment]
Don't use the mapping test. It's a pretty recent addition to jinja and not widely available.

**Comment by noelmcloughlin** [comment]
Fixed by #78
